src/riscv-legacy-integration.adoc

```diff
@@ -250,13 +250,33 @@ cause illegal instruction exceptions
 addresses (xref:csr-numbers-section[xrefstyle=short]) cause illegal
 instruction exceptions
 * All allowed instructions execute as if the CHERI execution mode is Legacy.
-The CME bits in <<mseccfg>>, <<menvcfg>>, and <<senvcfg>> have no effect whilst
+The CME bits in <<menvcfg>> and <<senvcfg>> are zero whilst
 CHERI register access is disabled.
 
 CHERI register access is disabled if XLEN in the current mode is less than
 XLENMAX or if CRE active at the current mode (<<menvcfg>>.CRE for S-mode or
 <<senvcfg>>.CRE for U-mode) is 0.
 
+<<menvcfg>>.CRE, <<menvcfg>>.CME, <<senvcfg>>.CRE and <<senvcfg>>.CME form a
+single WARL field. This allows higher privilege software to restrict lower
+privilege software access to CHERI register state, and the ability to enter
+capability mode. The valid configurations are shown in
+xref:xenvcfg-warl-field[xrefstyle=short].
+
+[[xenvcfg-warl-field]]
+.Xenvcfg joint WARL field
+[%autowidth,float="center",align="center",cols="<,<,<,<,<",options="header"]
+|===
+|<<menvcfg>>.CRE|<<menvcfg>>.CME|<<senvcfg>>.CRE|<<senvcfg>>.CME|Comment
+|0              |0              |0              |0              |<<menvcfg>>.CRE=0 disables all other state
+|1              |0/1            |0              |0              |<<senvcfg>>.CRE=0 disables <<senvcfg>>.CME
+|1              |0/1            |1              |0/1            |both levels of CME can be programmed
```

Comment:
I find this table slightly difficult to parse, maybe the following makes the implications of setting a given bit more obvious?
```suggestion
|0              | RO 0          | RO 0          | RO 0          |<<menvcfg>>.CRE=0 disables all other state
|1              |0/1            |0              | RO 0          |<<senvcfg>>.CRE=0 disables <<senvcfg>>.CME
|1              |0/1            |1              |0/1            |both levels of CME can be programmed
```